### PR TITLE
[Merged by Bors] - feat Port/data.list.big operators.lemmas

### PR DESCRIPTION
**Pull request by casavaca**
- [x] depends on #1380

### CODE REVIEW

**Comment by casavaca** [comment]
Should work after #1380 is merged.

**Comment by casavaca** [comment]
@semorrison why the label `blocked-by-other-PR` is removed?

File: Mathlib/Data/List/BigOperators/Lemmas.lean [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import Mathlib.Algebra.GroupWithZero.Commute
+import Mathlib.Algebra.GroupWithZero.Divisibility
+import Mathlib.Algebra.Order.WithZero
+import Mathlib.Algebra.Ring.Basic
+import Mathlib.Algebra.Ring.Divisibility
+import Mathlib.Algebra.Ring.Commute
+import Mathlib.Data.Int.Units
+import Mathlib.Data.Set.Basic
+
+/-! # Lemmas about `list.sum` and `list.prod` requiring extra algebra imports -/
```
Comment (jcommelin) [inline Mathlib/Data/List/BigOperators/Lemmas.lean:23]:
```suggestion
/-! # Lemmas about `List.sum` and `List.prod` requiring extra algebra imports -/
```

File: Mathlib/Data/List/BigOperators/Lemmas.lean [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  · rw [List.sum_cons]
+    exact dvd_add (h _ (mem_cons_self _ _)) (ih fun x hx => h x (mem_cons_of_mem _ hx))
+#align list.dvd_sum List.dvd_sum
+
+section Alternating
+
+variable [CommGroup α]
+
+@[to_additive]
+theorem alternating_prod_append :
```
Comment (jcommelin) [inline Mathlib/Data/List/BigOperators/Lemmas.lean:105]:
```suggestion
theorem alternatingProd_append :
```

File: Mathlib/Data/List/BigOperators/Lemmas.lean [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    ∀ l₁ l₂ : List α,
+      alternatingProd (l₁ ++ l₂) = alternatingProd l₁ * alternatingProd l₂ ^ (-1 : ℤ) ^ length l₁
+  | [], l₂ => by simp
+  | a :: l₁, l₂ => by
+    simp_rw [cons_append, alternating_prod_cons, alternating_prod_append, length_cons, pow_succ,
+      neg_mul, one_mul, zpow_neg, ← div_eq_mul_inv, div_div]
+#align list.alternating_prod_append List.alternating_prod_append
+
+@[to_additive]
+theorem alternating_prod_reverse :
```
Comment (jcommelin) [inline Mathlib/Data/List/BigOperators/Lemmas.lean:115]:
```suggestion
theorem alternatingProd_reverse :
```

File: Mathlib/Data/List/BigOperators/Lemmas.lean [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+theorem unop_map_list_prod {F : Type _} [MonoidHomClass F M Nᵐᵒᵖ] (f : F) (l : List M) :
+    (f l.prod).unop = (l.map (MulOpposite.unop ∘ f)).reverse.prod := by
+  rw [map_list_prod f l, MulOpposite.unop_list_prod, List.map_map]
+#align unop_map_list_prod unop_map_list_prod
+
+namespace MonoidHom
+
+/-- A morphism into the opposite monoid acts on the product by acting on the reversed elements.
+
+Deprecated, use `_root_.unop_map_list_prod` instead. -/
```
Comment (jcommelin) [inline Mathlib/Data/List/BigOperators/Lemmas.lean:174]:
```suggestion
/-- A morphism into the opposite monoid acts on the product by acting on the reversed elements. =/
@[deprecated _root_.unop_map_list_prod]
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**